projects/control-service/projects/base/src/main/java/com/vmware/taurus/service/diag/HttpTracer.java

```diff
@@ -117,27 +117,29 @@ void sendTelemetry(HttpTrace trace) {
     ObjectMapper mapper = new ObjectMapper();
     mapper.registerModule(new JavaTimeModule());
     ObjectNode node = mapper.convertValue(trace, ObjectNode.class);
-    node.put("@type", "taurus_httptrace");
-    node.put("deployment_mode", String.join(",", environment.getActiveProfiles()));
-    String opId = operationContext.getOpId();
-    if (null != opId) {
-      node.put("@id", opId);
-    }
-    HttpServletRequest request = operationContext.getRequest();
-    if (null != request) { // methods from the actuator dont have the request set
-      String requestMapping =
-          "" + request.getAttribute(HandlerMapping.BEST_MATCHING_PATTERN_ATTRIBUTE);
-      node.put("request_mapping", requestMapping);
-    }
-    String user = operationContext.getUser();
-    if (null != user) {
-      node.put("user", user);
-    }
-    String team = operationContext.getTeam();
-    if (null != team) {
-      node.put("team", team);
+    if (node != null) {
```

Comment:
This only returns null if httpTrace is null? 
Maybe we could add a @NotNull annotation on the parameter and maybe an assertion instead of doing this?

Reply:
The node object is a result of the conversion of the trace object by the object mapper and according to the coverity scan and the object mapper java doc, this method could still return a null value even if the argument is not null, hence this check.

Reply:
Where in the docs does it say that it can return null for a non null object ? these are the docs I'm looking at https://fasterxml.github.io/jackson-databind/javadoc/2.8/com/fasterxml/jackson/databind/ObjectMapper.html#convertValue(java.lang.Object,%20java.lang.Class)

Reply:
I agree.. sorry... I looked at some examples of edge cases where either the object was null or was of primitive type, which cannot happen at this part of the code. 

btw, we are using a newer version of the library and the correct javadoc link is: https://fasterxml.github.io/jackson-databind/javadoc/2.13/com/fasterxml/jackson/databind/ObjectMapper.html#convertValue(java.lang.Object,%20java.lang.Class)